lib/adapter/external/manga_kakalot.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Ewbscp.Adapter.External.MangaKakalot do
+  use Tesla
+
+  plug(Tesla.Middleware.BaseUrl, "https://ww7.mangakakalot.tv")
+  plug(Tesla.Middleware.Headers, [{"authorization", "token xyz"}])
+  plug(Tesla.Middleware.JSON)
+
+  # returns the page with the search results (this would be a list of links to be presented to the user)
+  def search_manga(manga_title) do
+    case get("/search/" <> manga_title) do
+      {:ok, %{status: 200, body: body}} ->
+        {:ok, body}
+
+      {:error, reason} ->
+        {:error, reason}
+
+      _ ->
+        {:error, :unknown}
+    end
+  end
+
+  def get_chapter_pages() do
```

Comment:
The idea is to use this fn in order to fetch the links retrieved by the previous fn?

Reply:
Initially, yes, but I was thinking about the project structure and thought that maybe this function belongs elsewhere.